app/controllers/pages_controller.rb

```diff
@@ -1,9 +1,24 @@
 class PagesController < ApplicationController
+  before_action :authenticate_user!, only: :date_chart
+  before_action :search_transactions, only: %i[home date_chart]
+
   def home
     return unless user_signed_in?
     @form = BalanceTransactionForm.new(Transaction.new, balance_transactions: BalanceTransaction.new)
     @last_transactions = current_user.transactions_last_ten
     @categories = current_user.categories.main_category
-    @date = Charts::Home::CategoriesDate.new(current_user.categories_have_budget).date
+    @date = Charts::Home::GetData.new(@transactions).call
+  end
+
+  def date_chart
+  end
+
+  private
+
+  def search_transactions
+    return unless user_signed_in?
```

Comment:
@marderer what do you think about removing this line and add `authenticate_user!` action before `#home` too?

```ruby
before_action :authenticate_user!
before_action :search_transactions
```